### PR TITLE
behave: 1.2.5 -> 1.2.6

### PR DESCRIPTION
**Pull request by alunduil**
###### Motivation for this change

Simple package bump of behave.  This also includes bumps of parse-type and parse to preserve
dependencies of behave.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/parse-type/default.nix [hunk 1]
```diff
@@ -6,21 +6,13 @@
 
 buildPythonPackage rec {
   pname = "parse_type";
-  version = "0.3.4";
+  version = "0.4.2";
```
Comment (Mic92) [inline pkgs/development/python-modules/parse-type/default.nix]: Can you make this a separate commit?

File: pkgs/development/python-modules/parse/default.nix [hunk 2]
```diff
@@ -3,22 +3,14 @@
 }:
 buildPythonPackage rec {
   pname = "parse";
-  version = "1.6.6";
+  version = "1.8.2";
```
Comment (Mic92) [inline pkgs/development/python-modules/parse/default.nix]: Can you make this a separate commit?

**Review by Mic92 — COMMENTED** [review]
Also please change to commit messages to have the same format as the pull request title (that I changed).

**Comment by alunduil** [comment]
I cleaned up those commits and made sure each package was separate.  Let me know if there is anything else that needs to be changed.

**Comment by Mic92** [comment]
@GrahamcOfBorg build python27Packages.parse python36Packages.parse-type  python27Packages.pyparser python36Packages.parse python36Packages.pyparser python27Packages.behave papis python27Packages.parse-type

**Comment by Mic92** [comment]
Looks like `traceback2` is missing for python2 or so?

**Comment by alunduil** [comment]
Since behave is designed to be a standalone executable, I've disabled py27.  This should resolve the build issues that came up. Let me know if there is anything else.

File: pkgs/development/python-modules/behave/default.nix [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 , glibcLocales, parse, parse-type, six
 }:
 buildPythonApplication rec {
   pname = "behave";
-  version = "1.2.5";
+  version = "1.2.6";
   name = "${pname}-${version}";
 
-  disabled = pythonAtLeast "3.6";
+  disabled = isPy27;
```
Comment (Mic92) [inline pkgs/development/python-modules/behave/default.nix]: So this is not supposed to be executed with the python version one wants to run the tests with? https://github.com/behave/behave

Comment (alunduil) [reply pkgs/development/python-modules/behave/default.nix]: You are correct and I'll add python27 back in.  Thanks for pointing out a bad solution.

On a slightly tangential note, how much longer will python27 be the default python installation and available on NixOS?

**Comment by alunduil** [comment]
I've fixed the unacceptable build error and have verified that python27 does build correctly now.

**Comment by Mic92** [comment]
@alunduil regarding python3 https://github.com/NixOS/nixpkgs/issues/18185

**Comment by Mic92** [comment]
@GrahamcOfBorg build python27Packages.parse python36Packages.parse-type python27Packages.pyparser python36Packages.parse python36Packages.pyparser python27Packages.behave papis python27Packages.parse-type

**Comment by Mic92** [comment]
Was this `bibtexparser` broken before? 

```console
$ nix-shell -p papis
...
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/bwriter.py -> build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/bparser.py -> build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/bibdatabase.py -> build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/__init__.py -> build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/latexenc.py -> build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/customization.py -> build/bdist.linux-x86_64/wheel/bibtexparser
copying build/lib/bibtexparser/bibtexexpression.py -> build/bdist.linux-x86_64/wheel/bibtexparser
running install_egg_info
running egg_info
writing bibtexparser.egg-info/PKG-INFO
writing dependency_links to bibtexparser.egg-info/dependency_links.txt
writing requirements to bibtexparser.egg-info/requires.txt
writing top-level names to bibtexparser.egg-info/top_level.txt
reading manifest file 'bibtexparser.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.md'
writing manifest file 'bibtexparser.egg-info/SOURCES.txt'
Copying bibtexparser.egg-info to build/bdist.linux-x86_64/wheel/bibtexparser-1.0.1-py3.6.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/bibtexparser-1.0.1.dist-info/WHEEL
creating '/build/bibtexparser-1.0.1/dist/bibtexparser-1.0.1-py3-none-any.whl' and adding '.' to it
adding 'bibtexparser/__init__.py'
adding 'bibtexparser/bibdatabase.py'
adding 'bibtexparser/bibtexexpression.py'
adding 'bibtexparser/bparser.py'
adding 'bibtexparser/bwriter.py'
adding 'bibtexparser/customization.py'
adding 'bibtexparser/latexenc.py'
adding 'bibtexparser-1.0.1.dist-info/DESCRIPTION.rst'
adding 'bibtexparser-1.0.1.dist-info/metadata.json'
adding 'bibtexparser-1.0.1.dist-info/top_level.txt'
adding 'bibtexparser-1.0.1.dist-info/WHEEL'
adding 'bibtexparser-1.0.1.dist-info/METADATA'
adding 'bibtexparser-1.0.1.dist-info/RECORD'
installing
/build/bibtexparser-1.0.1/dist /build/bibtexparser-1.0.1
Processing ./bibtexparser-1.0.1-py3-none-any.whl
Collecting pyparsing (from bibtexparser==1.0.1)
  Could not find a version that satisfies the requirement pyparsing (from bibtexparser==1.0.1) (from versions: )
No matching distribution found for pyparsing (from bibtexparser==1.0.1)
builder for '/nix/store/axil5yxwcvapd4nirvnb8bbajp8865lq-python3.6-bibtexparser-1.0.1.drv' failed with exit code 1
cannot build derivation '/nix/store/h0q3lz09kdjyc9mw8l6aj76ypb88m8gc-papis-0.5.3.drv': 1 dependencies couldn't be built
error: build of '/nix/store/h0q3lz09kdjyc9mw8l6aj76ypb88m8gc-papis-0.5.3.drv' failed
```

Looks like an easy fix given we already have pyparsing though.

**Comment by alunduil** [comment]
@Mic92 I don't think the bibtexparser issue was there before.  Let me know if you'd like me to take a crack at fixing that in this branch.

**Comment by Mic92** [comment]
I would be glad, if you could take a look at it @alunduil

**Comment by alunduil** [comment]
@Mic92 I've fixed the bibtexparser build issues (verified with a `nix-build -A papis` run locally).  Let me know if the commit message is acceptable or if it should be changed to something else.  Hopefully, the next run will show everything working as expected.

**Comment by alunduil** [comment]
Finally got a moment to resolve the conflicts.  Let me know if any of my choices should be different.

**Comment by Mic92** [comment]
Thanks!
